hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/StructuredTraceGraphBuilder.java

```diff
@@ -0,0 +1,47 @@
+package org.hypertrace.traceenricher.trace.util;
+
+import org.hypertrace.core.datamodel.StructuredTrace;
+import org.hypertrace.core.datamodel.shared.StructuredTraceGraph;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class StructuredTraceGraphBuilder {
+  private static final Logger LOG = LoggerFactory.getLogger(StructuredTraceGraphBuilder.class);
+
+  private static ThreadLocal<StructuredTraceGraph> cachedGraph = new ThreadLocal<>();
```

Comment:
Why do we want it specific to a Thread? Can we use Supplier instead?

Reply:
Currently, StructureTraceGraph's state is changing while processing a trace, and we want to re-construct whenever the state affecting its changes. So, we choose thread-local to share the state in a given kstream thread. 
@s what do you mean by supplier here? Are you referring to supplying StructureTraceGraph at the start of processing of topology? 
cc: @laxmanchekka

Reply:
@kotharironak We use Suppliers in a similar way to cache the results https://mkyong.com/java8/java-8-supplier-examples/

Though I am not sure, if you can change the entry in supplier based on a certain condition, if the graph structure can potentially change